app/controllers/support_ticket/check_your_request_controller.rb

```diff
@@ -2,15 +2,18 @@ class SupportTicket::CheckYourRequestController < SupportTicket::BaseController
   before_action :require_support_ticket_data!, only: :new
 
   def new
-    @form = form
+    @form ||= SupportTicket::CheckYourRequestForm.new(support_ticket: support_ticket)
     @school_details_path = school_details_path
     render 'support_tickets/check_your_request'
   end
 
   def save
-    if form.valid? && form.create_ticket
-      session[:support_ticket] = nil
-      session[:support_ticket_number] = form.ticket_number
+    @form ||= SupportTicket::CheckYourRequestForm.new(support_ticket: support_ticket)
+
+    if @form.valid? && @form.create_ticket
+      support_ticket.destroy!
```

Comment:
I'm not sure I follow the logic here - it's probably a problem with me rather than your code, but could you explain why you destroy the `support_ticket` if `@form.create_ticket` is true?

Reply:
the old code didn't quite have things in the right place. essentially `create_ticket` sends to zendesk then `destroy!` is called as we no longer need to persist this temporary data. I will:
- move `create_ticket` to `SupportTicket` and maybe rename
- `SupportTicket::CheckYourRequestForm` isn't really form so i may kill that and replace with just `SupportTicket`

Reply:
I did the first bullet point but decided to keep the form object, despite the fact it hardly does anything just to keep the similarity to the other pages